eclair-core/src/test/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWalletSpec.scala

```diff
@@ -141,10 +141,10 @@ class BitcoinCoreWalletSpec extends TestKit(ActorSystem("test")) with BitcoindSe
     sender.send(bitcoincli, BitcoinReq("getrawtransaction", fundingTxes(2).txid.toString()))
     assert(sender.expectMsgType[JString](10 seconds).s === fundingTxes(2).toString())
 
-    // NB: bitcoin core doesn't clear the locks when a tx is published
+    // NB: bitcoin core before 0.17.0 doesn't clear the locks when a tx is published
     sender.send(bitcoincli, BitcoinReq("listlockunspent"))
-    assert(sender.expectMsgType[JValue](10 seconds).children.size === 2)
-
+    val expectedLocks = if (bitcoinVersion >= 170000) 0 else 2
```

Comment:
We shouldn't have things like `if (bitcoinVersion >= 170000)`: our test runs against a certain version of bitcoin, backward compatibility isn't in scope here.